src/frontend/src/app/pages/open-proposal-details/open-proposal-details.component.ts

```diff
@@ -0,0 +1,214 @@
+import { CommonModule, DatePipe } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnInit,
+  SecurityContext,
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { DomSanitizer } from '@angular/platform-browser';
+import { ActivatedRoute, Router } from '@angular/router';
+import { marked } from 'marked';
+import { filter, map } from 'rxjs';
+
+import { ProposalLinkBaseUrl, ProposalState, ProposalTopic } from '~core/state';
+import { ProposalService } from '~core/state';
+import {
+  CardComponent,
+  KeyColComponent,
+  KeyValueGridComponent,
+  ValueColComponent,
+} from '~core/ui';
+import { isNotNil } from '~core/utils';
+
+@Component({
+  selector: 'app-open-proposal-details',
+  standalone: true,
+  imports: [
+    CommonModule,
+    CardComponent,
+    KeyValueGridComponent,
+    KeyColComponent,
+    ValueColComponent,
+    DatePipe,
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styles: [
+    `
+      @import '@cg/styles/common';
+
+      .proposal {
+        margin-bottom: size(6);
+      }
+
+      .proposal-title {
+        word-break: break-all;
+      }
+
+      :host {
+        display: block;
+        margin-left: auto;
+        margin-right: auto;
+
+        @include lg {
+          width: 75%;
+        }
+
+        @include xl {
+          width: 66.66667%;
+        }
+      }
+
+      .proposal-links__link {
+        margin-right: size(4);
+      }
+
+      .proposal-summary {
+        margin-top: size(3);
+        padding-top: size(6);
+        border-top: 1px solid $slate-300;
+
+        @include dark {
+          border-color: $slate-700;
+        }
+      }
+
+      .proposal-summary__content {
+        background-color: $slate-200;
+
+        @include dark {
+          background-color: $slate-900;
+        }
+      }
+    `,
+  ],
+  template: `
+    @if (currentProposal$ | async; as proposal) {
+      <h2 class="h3">{{ proposal.title }}</h2>
+
+      <app-card class="proposal">
+        <app-key-value-grid [columnNumber]="2">
+          <app-key-col id="proposal-id">ID</app-key-col>
+          <app-value-col aria-labelledby="proposal-id">
+            <a
+              href="{{ linkBaseUrl.ProposalId }}{{ proposal.id }}"
+              target="_blank"
+              rel="nofollow noreferrer"
+              >{{ proposal.id }}
+            </a>
+          </app-value-col>
+
+          <app-key-col id="proposal-links">Voting links</app-key-col>
+          <app-value-col
+            class="proposal-links"
+            aria-labelledby="proposal-links"
+          >
+            @if (proposal.proposalLinks !== []) {
+              @for (
+                proposalLink of proposal.proposalLinks;
+                track proposalLink.type
```

Comment:
`track` should take a value that is unique to every entry, such as the ID. It uses this value to determine whether or not the element should re-render. Using `type`, if the specific proposal in a list of proposals was replaced with a new proposal that had the same type then it would not re-render.